### PR TITLE
Add sync state to walletinfo

### LINKED ISSUE #2564: Add state descriptor to wallet info rpc
This would be extremely useful to have on this endpoint. Basically include the block height and block hash along with all of our other information. I was going to use this rpc as an example for #2486

```
 ./bitcoin-s-cli walletinfo
{
  "wallet": {
    "keymanager": {
      "rootXpub": ...
    },
    "xpub": ...,
    "hdPath": "m/84'/0'/0'",
    "height" : 0,
    "blockHash" : "1234....567890"
  }
}
```

### PR DESCRIPTION
**Pull request by benthecarman**
Fixes #2564

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala [hunk 1]
```diff
@@ -396,6 +396,8 @@ trait WalletApi extends StartStopAsync[WalletApi] {
     * is a change output from this wallet
     */
   def isChange(output: TransactionOutput): Future[Boolean]
+
+  def getSyncState(): Future[(Int, DoubleSha256DigestBE)]
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala]: So this LGTM, but one thing that might be nice is to rename our current `FilterSyncMarker` to just `SyncMarker` and then use that here too? 

https://github.com/bitcoin-s/bitcoin-s/blob/2de17bb4e47961fd3fe4e55d75909b42470e3c59/core/src/main/scala/org/bitcoins/core/api/chain/FilterSyncMarker.scala#L11

Comment (Christewart) [reply core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala]: It looks like we've had this discussion before, are the contexts the same? https://github.com/bitcoin-s/bitcoin-s/pull/2003#issuecomment-691176472

Comment (benthecarman) [reply core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala]: I don't think it makes sense to use `FilterSyncMarker` or renaming it. `FilterSyncMarker`'s height and hash aren't meant to refer to the same block

Comment (Christewart) [reply core/src/main/scala/org/bitcoins/core/api/wallet/WalletApi.scala]: ok, then possibly make a `BlockSyncMarker` with appropriate field names?

**Review by Christewart — APPROVED** [review]

**Pull request merged**